hummingbot/connector/connector_base.pyx

```diff
@@ -264,6 +266,47 @@ cdef class ConnectorBase(NetworkIterator):
         """
         raise NotImplementedError
 
+    def batch_order_create(self, orders_to_create: List[LimitOrder]) -> List[LimitOrder]:
+        """
+        Issues a batch order creation as a single API request for exchanges that implement this feature. The default
+        implementation of this method is to send the requests discretely (one by one).
+        :param orders_to_create: A list of LimitOrder objects representing the orders to create. The order IDs
+            can be blanc.
+        :returns: A tuple composed of LimitOrder objects representing the created orders, complete with the generated
+            order IDs.
+        """
+        creation_results = []
+        for order in orders_to_create:
+            if order.is_buy:
+                client_order_id = self.buy(
+                    trading_pair=order.trading_pair,
+                    amount=order.quantity,
+                    order_type=OrderType.LIMIT,
+                    price=order.price,
+                )
```

Comment:
Is there a reason why the default behaviour for `batch_order_create` is in contrast to what it is originally meant to do?

Reply:
Yes @CrimsonJacket . This is the default implementation we add in ConnectorBase to make all connectors polymorphic with the new functionality. But we are not providing now the logic to the the batch actions in all connectors, only for Injective. That is why the default behavior has to be doing the same thing we do when requested actions for a single order: handling them one by one.

Reply:
> Yes @CrimsonJacket . This is the default implementation we add in ConnectorBase to make all connectors polymorphic with the new functionality. But we are not providing now the logic to the the batch actions in all connectors, only for Injective. That is why the default behavior has to be doing the same thing we do when requested actions for a single order: handling them one by one.

Hello, I see you have been active in certain areas where we need some simple but urgent help.  Can you please contact me on telegram @saveplanetearthdev , thank you.